File: Cpp/2586_count_the_number_of _vowel_strings_in_range.cpp
// https://leetcode.cn/problems/count-the-number-of-vowel-strings-in-range/

class Solution {
public:
    int vowelStrings(vector<string>& words, int left, int right) {
        int res = 0;
        unordered_set<char> vowels = {'a', 'e', 'i', 'o', 'u'};
        for (int i = left; i <= right; ++i) {
            if (vowels.find(words[i][0]) != vowels.end() &&
                vowels.find(words[i].back()) != vowels.end()) {
                res += 1;
            }
        }
        return res;
    }
};

File: Python3/2586_count_the_number_of_vowel_strings_in_range.py
# https://leetcode.cn/problems/count-the-number-of-vowel-strings-in-range/

class Solution:
    def vowelStrings(self, words, left, right):
        """
        Inputs:
            words: list[str]
            left, right: int
        Outputs:
            res: int
        """
        res = 0
        vowels = {"a", "e", "i", "o", "u"}
        for i in range(left, right+1):
            if words[i][0] in vowels and words[i][-1] in vowels:
                res += 1
        return res